stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt

```diff
@@ -30,6 +31,7 @@ public class ChannelListView @JvmOverloads constructor(
     private val layoutManager: ScrollPauseLinearLayoutManager
     private val scrollListener: EndReachedScrollListener = EndReachedScrollListener()
     private val dividerDecoration: SimpleVerticalListDivider = SimpleVerticalListDivider()
+    private var loadingMore = false
```

Comment:
This is duplicating state from the adapter, and might get out of sync if we ever forget to update it. We could check whether we are currently loading with something like `adapter.currentList.contains(LoadingMoreItem)`.

Reply:
That was how I had initially implemented it, but it seemed like searching a potentially large list of items to see if we're loading might have some performance impacts.

Reply:
I don't think performance will be an issue here, and if it is, we can always optimize it later.

- Channel lists probably won't be super long most of the time
- We don't need to do this often, only when the loading/pagination state changes
- Iterating an `ArrayList` should be very quick
- We only need to compare references, and not values, because we're looking for an `object`

Reply:
Fair enough!